core/src/main/java/kafka/automq/table/deserializer/proto/LastestSchemaResolutionResolver.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2025, AutoMQ HK Limited.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kafka.automq.table.deserializer.proto;
+
+import kafka.automq.table.deserializer.SchemaResolutionResolver;
+import kafka.automq.table.deserializer.proto.schema.MessageIndexes;
+
+import org.apache.kafka.common.errors.SerializationException;
+import org.apache.kafka.common.record.Record;
+
+import com.automq.stream.utils.Time;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.Collections;
+import java.util.concurrent.ConcurrentHashMap;
+
+import io.confluent.kafka.schemaregistry.client.SchemaMetadata;
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+import io.confluent.kafka.schemaregistry.client.rest.exceptions.RestClientException;
+
+/**
+ * Implementation of SchemaResolutionResolver that retrieves the latest schema from Schema Registry by subject name.
+ * This implementation includes caching mechanism to avoid frequent registry queries.
+ * Cache entries are refreshed every 5 minutes.
+ */
+public class LastestSchemaResolutionResolver implements SchemaResolutionResolver {
```

Comment:
If the LastestSchemaResolutionResolver is created per IcebergWrite, we need to consider the cost